### PR TITLE
Checks of XML format on enrollment

### PR DESCRIPTION
**Pull request by DIOHz0r**
### Changes description

Add additional XML parse tries to avoid format import problems on device enrolment

### Checklist

Please check if your PR fulfills the following specifications:

- [x] Tests for the changes have been added
- [ ] Docs have been added/updated

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +34,7 @@ before_install:
 before_script:
 - "./tests/before_script.sh"
 script:
-  - vendor/bin/atoum -ft -bf tests/bootstrap.php -d tests/suite-install -ncc
+  - vendor/bin/atoum -ft -bf tests/bootstrap.php -d tests/suite-install --ncc
```
Comment (btry) [inline .travis.yml]: this change is wrong

run ```atoum --help```

```
  -sf <file>, --score-file <file>                                           Save score in file <file>
  -mcn <integer>, --max-children-number <integer>                           Maximum number of sub-processes which will be run simultaneously
  -ncc, --no-code-coverage                                                  Disable code coverage
  -nccid <directory>..., --no-code-coverage-in-directories <directory>...   Disable code coverage in directories <directory>
```

File: tests/before_script.sh [hunk 2]
```diff
@@ -29,7 +29,7 @@ mkdir plugins/fusioninventory && git clone --depth=1 $FI_SOURCE -b $FI_BRANCH pl
 IFS=/ read -a repo <<< $TRAVIS_REPO_SLUG
 mv ../${repo[1]} plugins/flyvemdm
 cd plugins/fusioninventory
-patch -p1 < ../flyvemdm/tests/patches/fi-fix-obsolete-query.patch
+if [[ $FI_BRANCH == "glpi9.2+1.0" ]] ; then patch -p1 < ../flyvemdm/tests/patches/fi-fix-obsolete-query.patch; fi
```
Comment (btry) [inline tests/before_script.sh]: FI has a release 9.2+2.0. It is worth to run unit tests with this version and see if the patch may be dropped.

Comment (DIOHz0r) [reply tests/before_script.sh]: For that version of FI the query sentence hasn't been updated so the patch is working but for the master branch the code was updated and that patch is useless.

Comment (btry) [reply tests/before_script.sh]: OK

File: tests/patches/glpi-allow-test-on-master-branch.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/setup.php b/setup.php
+index eed6660..1f065eb 100644
+--- a/setup.php
++++ b/setup.php
+@@ -209,7 +209,7 @@ function plugin_version_flyvemdm() {
+       'requirements'   => [
+          'glpi' => [
+             'min' => $glpiVersion,
+-            'max' => PLUGIN_FLYVEMDM_GLPI_MAX_VERSION,
++            'max' => '9.5',
```
Comment (btry) [inline tests/patches/glpi-allow-test-on-master-branch.patch]: Funny you did not changed the constant itself :D This way, when GLPI 9.5 will begin, we may simply create a PR with the patch

**Pull request merged**